goawk.go

```diff
@@ -53,6 +54,9 @@ const (
   -v var=value      variable assignment (multiple allowed)
 
 Additional GoAWK arguments:
+  -covermode mode   set the coverage report mode (set/count)
+  -coverprofile file set the coverage report filename
+  -coverappend      do not truncate coverage report file
```

Comment:
No problem to remove the `-coverappend` switch at all, but in this case I propose to set the behavior to append. Why?

1. It's much easier to remove a cover profile file at start than to implement generating many files with different names and then merging. The more so is because you can't just concatenate the files, since the structure of coverage report is first line = mode line and other lines = data. 
2. _"If someone really wants it"_ - it's I am at first place who really wants it, as I've mentioned earlier. During testing of my awk code I run same awk program many times with different arguments and check the result. Thus I want to collect coverage from all those runs.

We will describe this behavior is documentation. Do you agree?

Reply:
Interesting idea. However, I don't like the default of append -- it seemed counterintuitive to me when I first tested your original PR, and it's different from the Go tooling (which we're somewhat tying ourselves to), so that seems a bit confusing too.

What about this: could we make the modes `set`, `append`, or `count`? How often would you want to append and count? (Does that even make sense?) Personally I've never used count in Go, but maybe you have.

Reply:
> This is quite a few arguments just for coverage (a niche feature). I suggest we get rid of the -coverappend argument and feature for now: Go doesn't have it either, and it seems like a niche sub-feature of a niche feature. :-) If someone really wants it ...
1. Go doesn't have it because Go is not AWK. In Go you can turn on coverage as part of tests run. So Go is suited to run many tests in single run. AWK being much simpler tool obviously lacks any kind of (built-in) testing framework. Thus having this feature allows to alleviate collecting coverage for AWK when you use [tush](https://github.com/adolfopa/tush)-style tests. 
2. Now when you propose to restructure the help message for goawk in https://github.com/benhoyt/goawk/pull/154#discussion_r996198318, I think it's perfect opportunity to just put the coverage switches in its own section. Thus it won't be such a big deal having one more option there.
3. Being able to collect code coverage statistics (using GoAWK) for my own projects (in AWK) was the main driving force and motivation for me for this effort. I'll be really upset if I only deliver the functionality barely useful for my use-cases.

Reply:
> I'll be really upset if I only deliver the functionality barely useful for my use-cases.

Yeah, I get that. Remember when we started I said we could take the easy way (do this on a branch) or the hard way (merge into master, but have thorough review). And you decided on the latter, so work with me here. :-) I'm not opposed to including the features you need, I just want to make sure the API is simple and clean and understandable for others, given that it's going to be in master.

It's true that with the debugging+coverage options in their own section it'll be less daunting.

> I'm not sure I completely understood the idea of adding this to modes (set, append, or count). To me these are logically distinct notions that probably should not be mixed.

I guess I haven't really used `count` at all, so don't know in what cases you'd use that. I was thinking that it'd be rare to use count+append, so how about we make `append` a mode instead (it would be equivalent to `set`+`append`). I realize that means we wouldn't support `count`+`append`, but maybe that's okay? Do you use count, and if so, what kinds of things do you use it for?

Reply:
Ok, now I've got your idea of making this a mode, thanks! However as I've mentioned I don't like that this mixes unrelated concepts, making this less clear and less flexible. If I would do it - I would prefer to leave this separate option as it is and placing in into relevant section of help string.

Reply:
Okay, that's fine. Let's go ahead with your original three args then: `-coverprofile file`, `-covermode set|count`, and `-coverappend`, and show them in the "GoAWK debugging options" section. If we come up with something cleaner later, we can always improve it (I don't think we need to be strictly backwards-compatible with these debugging commands).

Reply:
Fantastic! Thank you!